### PR TITLE
Align "Export Self" (`spec_export_customers`) implementation w/ text

### LINKED ISSUE #124: Customer cone size in AS stats correlations
https://publicdata.caida.org/datasets/as-relationships/serial-1/

### PR DESCRIPTION
**Pull request by SichangHe**
Instead of going through the customer cone, this implementation simply checks if the exported prefix is received from a customer AS.

Fix #124.

@cunha.

### CODE REVIEW

File: route_verification/bgp/src/cmp/filter.rs [hunk 1]
```diff
@@ -91,13 +91,25 @@ impl<'a> CheckFilter<'a> {
...
             && self.cmp.verbosity.check_customer
             && num == self.self_num
-            && self.filter_as_set(&customer_set(num), op).is_none()
+            && op.permits(&self.cmp.prefix)
+            && self.is_prev_as_customer()
+    }
+
+    /// Check if the previous ASN on the AS path is in our customer set.
+    fn is_prev_as_customer(&self) -> bool {
+        match self.last_on_path() {
```
Comment (cunha) [inline route_verification/bgp/src/cmp/filter.rs:106]: Does this return the origin AS or the previous AS (relative to the current AS) on the path? I think we want the previous AS on the path here.

**Comment by cunha** [comment]
I think we just need to check https://github.com/SichangHe/internet_route_verification/pull/193/files#r1719806711

File: route_verification/bgp/src/cmp/filter.rs (hunk 1 above)
Comment (SichangHe) [reply route_verification/bgp/src/cmp/filter.rs:106]: This returns the previous AS on the AS path relative to the AS under check (the exporting one, in this case).

**Comment by SichangHe** [comment]
@cunha, this should be it, right?

**Comment by SichangHe** [comment]
This renders all analysis that involves special cases stale. We will catch them when regenerating the artifacts.

**Pull request merged**